ubclaunchpad/pinpoint
# gateway: login

:tickets: **Ticket(s)**: Closes #

---

## :construction_worker: Changes

1. Added in cookie session to facilitate user login sessions (Universal-Cookie) in case user reloads browser
2. Fixed minor bug in retrieving user credentials at gateway from frontend side through unmarshalling which was causing crashes
3. Changed how response is passed back at the login function of index.js to prevent crashes when incorrect login
3. Moved log in functionality to higher level in react, App.js now managed loggedin state.

## :flashlight: Testing Instructions

1. Go to signup via navbar and create a new user
2. Try logging in with wrong password should result in no crashes and "incorrect login credential" being shown
3. Try logging in with correct credentials should redirect user to homepage with a navbar showing club pages
4. Can logout via logout button of navbar and delete the cookie session.
